pkg/controller/argocd/secret.go

```diff
@@ -379,6 +388,57 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	// do not create the secret if the key is not set.
+	if _, ok := os.LookupEnv("ARGOCD_CLUSTER_CONFIG_NAMESPACES"); !ok {
+		return nil
+	}
```

Comment:
I believe this environment variable will always be set. Shouldn't we rather make sure that we return when the current namespace is not in the list of namespaces exposed via this env var?

For example, if `ARGOCD_CLUSTER_CONFIG_NAMESPACES` is set to `openshift-gitops, argocd-cluster`, then we return with a no-op here if we are reconciling in either `openshift-gitops` or `argocd-cluster` namespace.

Reply:
Ah, hum, I might be mistaken here. So the intent here is that we only do this if the operator is being part of OpenShift GitOps, and that's when the variable will be set, correct?

Reply:
Yes right !! We are only using this check to distinguish between Community Operator and Openshift Gitop OPerator operations !!

Reply:
> I believe this environment variable will always be set. Shouldn't we rather make sure that we return when the current namespace is not in the list of namespaces exposed via this env var?

Yes.

We shouldn't have to inject any special downstream logic here. Whether "ARGOCD_CLUSTER_CONFIG_NAMESPACES" has "foo" or "openshift-gitops", it shouldn't matter.

Reply:
With this logic, and the environment _not_ set at all, the operator will currently:

* Set up namespace-scoped permissions for the `argocd-application-controller`
* _Not_ create a cluster secret to make the Argo CD instance namespace-scoped

resulting in an error.